common/tools/dev-tool/src/util/browserRelayServer.ts

```diff
@@ -47,7 +53,12 @@ function buildServer(app: Express) {
   app.put("/credential", (req, res) => {
     const id = randomUUID();
     try {
-      const cred = new DefaultAzureCredential(req.body);
+      const cred = new ChainedTokenCredential(
+        new AzurePowerShellCredential(req.body),
+        new AzureCliCredential(req.body),
+        // Keep Environment Credential for packages that have not migrated yet
+        new EnvironmentCredential(req.body),
```

Comment:
Environment Credential is supposed to be equivalent of ClientSecretCredential?

Reply:
IIUC Environment Credential is the superset and Client Secret Credential is created under the hood if tenant id, client id, and client secret are available in the environment. So yes, somewhat equivalent and should not break people. cc/ @KarishmaGhiya @maorleger please correct me if I misunderstood anything.

Reply:
That's [correct](https://github.com/Azure/azure-sdk-for-js/blob/4bf80049a0923cece80c8bfae0d740f0c279eb2c/sdk/identity/identity/src/credentials/environmentCredential.ts#L91-L97) 

Only caveat is ClientSecretCredential takes those values in as params, EnvironmentCredential expects them as env vars